// Copyright 2023 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package secretflow.spec.extend;

option java_package = "org.secretflow.spec.extend";

message ColumnQuery {
  enum AggregationFunction {
    INVAL = 0;
    SUM = 1;
    MEAN = 2;
    VAR = 3;
    MIN = 4;
    MAX = 5;
    COUNT = 6;
  }
  AggregationFunction function = 1;
  string column_name = 2;
}

message GroupbyAggregationConfig {
  repeated ColumnQuery column_queries = 1;
}